scripts/UTR.localdeploy.js

```diff
@@ -0,0 +1,29 @@
+const { ethers } = require("hardhat");
+
+const opts = {
+    gasLimit: 30000000
+};
+
+async function main() {
+    const [owner] = await ethers.getSigners();
+    const signer = owner;
+
+    // Setup to deploy UTR
+    // const url = "https://bsc-dataseed.binance.org/"
+    const singletonFactoryAddress = '0xce0042B868300000d44A59004Da54A005ffdcf9f';
+    const singletonFactoryABI = require('./abi/SingletonFactory.json');
+    const singletonFactory = new ethers.Contract(singletonFactoryAddress, singletonFactoryABI, owner.provider);
+    const initCodeUTR = require('@derivable/utr/build/UniversalTokenRouter.json').bytecode;
+    const salt = '0x46813c3ad3ccf666bcd988ce34c98823d067cd15e337d30d2ac76a189b112553';
```

Comment:
what is this `salt`?

Reply:
its saltHash
keccak256(salt)

Reply:
then you should write `ethers.utils.keccak256(salt)` for clarification